gpii/node_modules/eventLog/src/metrics.js

```diff
@@ -37,37 +37,74 @@ fluid.defaults("gpii.metrics", {
                     gradeNames: "gpii.windowsMetrics"
                 }
             }
-        },
-        metrics: {
-            checks: {
-                standaloneMetrics: {
-                    contextValue: "{gpii.contexts.standalone-metrics}"
-                }
-            },
-            defaultGradeNames: "gpii.metrics.lifecycle"
         }
     },
     invokers: {
         logMetric: {
             func: "{eventLog}.logEvent",
             args: ["metrics", "{arguments}.0", "{arguments}.1"] // event, data
+        },
+        startSubSession: {
+            funcName: "gpii.metrics.startSubSession",
+            args: ["{that}", "{eventLog}"]
+        },
+        stopSubSession: {
+            funcName: "gpii.metrics.stopSubSession",
+            args: ["{that}", "{eventLog}"]
         }
     },
     members: {
-        sessionSolutions: {}
+        sessionSolutions: {},
+        // Incrementing sub-session id.
+        subSessionIncrementer: 0
     },
     events: {
         "onStartMetrics": null,
-        "onStopMetrics": null
-    }
+        "onStopMetrics": null,
+        // The user has become active (called on the first input after onInactive)
+        "onActive": null, // args: duration-inactive
+        // The user has become inactive (no input was received after {that}.config.input.inactiveTime)
+        "onInactive": null // args: {sleep:true}
+    },
+    listeners: {
+        "onStartMetrics.log": {
+            funcName: "fluid.log",
+            args: "Metrics started"
+        },
+        "onActive": [{
```

Comment:
We prefer to use namespaced listeners, same applies to "onInactive"

Reply:
me too.. fixed